commit 1cc9212c059c4e76a1dae269a600cf7a84919a22 (from 430dcb65a7a8a34f3fdd6971e35678782abee4ea)
Merge: 430dcb6 a5f53f0
Author: ErikZalm <erik@vdzalm.eu>
Date:   Mon Feb 3 12:30:40 2014 -0800

    Merge pull request #744 from drf5n/watts
    
    Heater wattage reporting for M105 using EXTRUDER_WATTS and BED_WATTS

diff --git a/Marlin/Configuration.h b/Marlin/Configuration.h
index 1da62fc..fe03e03 100644
--- a/Marlin/Configuration.h
+++ b/Marlin/Configuration.h
@@ -156,6 +156,10 @@
 // HEATER_BED_DUTY_CYCLE_DIVIDER intervals.
 //#define HEATER_BED_DUTY_CYCLE_DIVIDER 4
 
+// If you want the M105 heater power reported in watts, define the BED_WATTS, and (shared for all extruders) EXTRUDER_WATTS 
+//#define EXTRUDER_WATTS (12.0*12.0/6.7) //  P=I^2/R 
+//#define BED_WATTS (12.0*12.0/1.1)      // P=I^2/R
+
 // PID settings:
 // Comment the following line to disable PID and enable bang-bang.
 #define PIDTEMP
diff --git a/Marlin/Marlin_main.cpp b/Marlin/Marlin_main.cpp
index d33f0ff..aecaf12 100644
--- a/Marlin/Marlin_main.cpp
+++ b/Marlin/Marlin_main.cpp
@@ -1816,10 +1816,20 @@ void process_commands()
       #endif
 
         SERIAL_PROTOCOLPGM(" @:");
+      #ifdef EXTRUDER_WATTS
+        SERIAL_PROTOCOL((EXTRUDER_WATTS * getHeaterPower(tmp_extruder))/127);
+        SERIAL_PROTOCOLPGM("W");
+      #else
         SERIAL_PROTOCOL(getHeaterPower(tmp_extruder));
+      #endif
 
         SERIAL_PROTOCOLPGM(" B@:");
+      #ifdef BED_WATTS
+        SERIAL_PROTOCOL((BED_WATTS * getHeaterPower(-1))/127);
+        SERIAL_PROTOCOLPGM("W");
+      #else
         SERIAL_PROTOCOL(getHeaterPower(-1));
+      #endif
 
         #ifdef SHOW_TEMP_ADC_VALUES
           #if defined(TEMP_BED_PIN) && TEMP_BED_PIN > -1

